src/main/java/org/auscope/portal/core/services/methodmakers/filter/AbstractFilter.java

```diff
@@ -394,22 +412,39 @@ protected String generatePropertyIsLessThan(String propertyName, String literal,
      * Will default to a 'case insensitive' match
      *
      * Will compare a property to see if it is greater than a literal
-     * 
+     *
      * @param propertyName
      *            The XPath to the property to compare
      * @param literal
      *            The literal to compare against
      * @return
      */
-    protected String generatePropertyIsGreaterThan(String propertyName, String literal) {
+    protected String generatePropertyIsGreaterThan(String propertyName, String literal ) {
         return generatePropertyIsGreaterThan(propertyName, literal, false, null);
     }
-
+    /**
+     * Generates an ogc:Filter string fragment that can be embedded in <ogc:And> <ogc:Or> <ogc:Not> <ogc:Filter> parent elements.
+     *
+     * Will default to a 'case insensitive' match and no attributes
+     *
+     * Will compare a property to see if it is greater than a literal
+     *
+     * @param propertyName
+     *            The XPath to the property to compare
+     * @param function
+     *            The function to compare against
+     * @return
+     */
+    protected String generatePropertyIsGreaterThan(String propertyName,String function,Boolean useFunction) {
+        if (useFunction == false)
+            return null;
+        return generatePropertyComparisonFragment("ogc:PropertyIsGreaterThan", null, propertyName, null, function);
+    }

```

Comment:
why are we using useFunction to overload this?

protected String generatePropertyIsGreaterThan(String propertyName, String literal, Boolean matchCase,
           MatchActionType matchAction, String function){
     //there should only be one method that calls generatePropertyComparisonFragment
}

protected String generatePropertyIsGreaterThan(String propertyName, String literal, Boolean matchCase,
           MatchActionType matchAction){
     generatePropertyIsGreaterThan(propertyName,literal,matchCase,matchAction,null);
}

Alternatively you can create another function e.g generateDatePropertyIsGreaterThan instead of overloading the param. This might be a better solution.